components/remote_settings/src/client.rs

```diff
@@ -102,22 +101,27 @@ impl Client {
     /// Downloads an attachment from [attachment_location]. NOTE: there are no
     /// guarantees about a maximum size, so use care when fetching potentially
     /// large attachments.
-    pub fn get_attachment(&self, attachment_location: &str) -> Result<Response> {
-        let mut current_remote_state = self.remote_state.lock();
-        self.ensure_no_backoff(&mut current_remote_state.backoff)?;
-        let attachments_base_url = match &current_remote_state.attachments_base_url {
-            Some(url) => url.clone(),
-            None => {
-                let req = Request::get(self.base_url.clone());
-                let server_info = req.send()?.json::<ServerInfo>()?;
-                match server_info.capabilities.attachments {
-                    Some(capability) => Url::parse(&capability.base_url)?,
-                    None => Err(RemoteSettingsError::AttachmentsUnsupportedError)?,
-                }
-            }
+    pub fn get_attachment(&self, attachment_location: &str) -> Result<Vec<u8>> {
+        Ok(self.get_attachment_raw(attachment_location)?.body)
+    }
+
+    /// Fetches a raw network [Response] for an attachment.
+    pub fn get_attachment_raw(&self, attachment_location: &str) -> Result<Response> {
+        if let Some(attachments_base_url) = self.remote_state.lock().attachments_base_url.clone() {
```

Comment:
This confused me a little - is it possible to match on the value in the state, so we don't early return here but still fall through to the `make_request` at the bottom?